traffic/traffic_info.cpp

```diff
@@ -87,6 +127,278 @@ TrafficInfo TrafficInfo::BuildForTesting(Coloring && coloring)
 
 bool TrafficInfo::ReceiveTrafficData()
 {
+  vector<SpeedGroup> values;
+  if (!ReceiveTrafficValues(values))
+    return false;
+
+  if (m_keys.size() != values.size())
+  {
+    LOG(LWARNING,
+        ("The number of received traffic values does not correspond to the number of keys:",
+         m_keys.size(), "keys", values.size(), "values."));
+    m_availability = Availability::NoData;
+    m_coloring.clear();
+    return false;
+  }
+
+  for (size_t i = 0; i < m_keys.size(); ++i)
+    m_coloring.emplace(m_keys[i], values[i]);
```

Comment:
Please, add a test when ReceiveTrafficData() is called twice, and for the same set of key there's a completely different set of values.